--- ./sys/targets/targets.go	2021-01-23 16:17:40.841742761 +0000
+++ ./sys/targets/targets.go	2021-01-23 16:14:25.928603917 +0000
@@ -35,7 +35,7 @@
 	KernelHeaderArch string
 	BrokenCompiler   string
 	// NeedSyscallDefine is used by csource package to decide when to emit __NR_* defines.
-	NeedSyscallDefine  func(nr uint64) bool
+	NeedSyscallDefine  func(nr uint64) bool `json:"-"`
 	HostEndian         binary.ByteOrder
 	SyscallTrampolines map[string]string
 
