--- ./executor/executor.cc	2021-02-06 14:12:41.368275378 +0000
+++ ./executor/executor.cc	2021-02-06 14:11:37.559564000 +0000
@@ -360,6 +360,8 @@
 #endif
 
 #include "cov_filter.h"
+#include "ivshm_setup.h"
+#include "features.h"
 
 #include "test.h"
 
@@ -392,12 +394,15 @@
 	if (argc == 2 && strcmp(argv[1], "test") == 0)
 		return run_tests();
 
+	FEATURES_CHECK_SNIPPET;
+
 	start_time_ms = current_time_ms();
 
 	os_init(argc, argv, (char*)SYZ_DATA_OFFSET, SYZ_NUM_PAGES * SYZ_PAGE_SIZE);
 	current_thread = &threads[0];
 
 #if SYZ_EXECUTOR_USES_SHMEM
+	IVSHM_SETUP_SNIPPET;
 	if (mmap(&input_data[0], kMaxInput, PROT_READ, MAP_PRIVATE | MAP_FIXED, kInFd, 0) != &input_data[0])
 		fail("mmap of input file failed");
 	// The output region is the only thing in executor process for which consistency matters.
